### ISSUE TITLE
Deployment of Presearch node in jukebox doesn't appear in presearch dashboard

### ISSUE THREAD

**Issue by gmachtel** [threefoldtech/tf_jukebox#85]
I tried to reserve a PRE node, filled in the code. Chatflow mentioned all finished well but : 
- I can't see the dashboard on our side like I do in VDC
<img width="1216" alt="Screenshot 2021-09-02 at 12 54 20" src="https://user-images.githubusercontent.com/30384423/131831855-3c21f8ce-6e62-495e-8330-7af0e1ee1b1d.png">

- In Presearch dashboard my node doesn't appear. It does when I launch it from VDC. 
<img width="1216" alt="Screenshot 2021-09-02 at 12 55 42" src="https://user-images.githubusercontent.com/30384423/131831891-b56c0a6b-d8f5-420f-afa6-2bea5c111a20.png">

**Comment by ashraffouda** [comment]
@abdulgig  is investigating this

**Comment by abdulgig** [comment]
So here is the thing, in the VDC we kill the pod and it redeploys a fresh image of PRE from the docker repo

In case of JukeBox, we need to somehow update the flists in the same way. We'll need to find a way so that JukeBox gets the updated flist , plus we may need some sort of conversion script that converts a docker image to an flist and then pushes it to a location from JukeBox can use it to deploy containers / vms . .

**Comment by ashraffouda** [comment]
https://github.com/threefoldtech/tf_jukebox/issues/86

**Comment by abdulgig** [comment]
@gmachtel Can you please try to deploy again ? It should be fine now as we just tested

![image](https://user-images.githubusercontent.com/25789764/132127551-d979c9bb-c825-4f37-8a61-161381917bfa.png)

**Comment by gmachtel** [comment]
I deleted previous version, and reinstalled current one. But it seems that the IPv4 address is still connected. As I don't see it in the presearch node dashboard, and in the logs I see a duplicate IP. Something remained hanging from my old deployment? 
You can only run one PRE node under 1 IPv4 address. 
<img width="1262" alt="Screenshot 2021-09-06 at 10 38 22" src="https://user-images.githubusercontent.com/30384423/132187458-a71e06a2-3670-4507-8f35-be29d517003d.png">

**Comment by abdulgig** [comment]
At this point, the containers only have an IPv6 address, not a public IPv4 address. 

Presearch doesn't support IPv6, only IPv4. So it picks up the farm's egress IPv4 and registers itself with Presearch network.  So most probably you can have 1 node per farm. Hopefully under grid 3.0 we can have workloads/VMs with public IPv4 addresses that should resolve this issue.

**Comment by gmachtel** [comment]
OK but in the meantime in jukebox it doesn't work? Then we can't present it to them...

**Comment by abdulgig** [comment]
With the aforementioned issue, it won't unless you deploy it on a different farm. For demonstration purposes, we can use the solution in the VDC as that is stable

**Comment by gmachtel** [comment]
There is something I don’t understand: there is no Presearch deployment on this farm, at least not one that I created. I used the jukebox, don’t know which farm is used for this, as this info is hidden.

**Comment by ashraffouda** [comment]
I think we should disable it if it is not working properly

**Comment by gmachtel** [comment]
I agree if it doesn’t work, but I first want to understand what’s wrong with the setup. Are all the jukebox deployments on the same farm?

Be aware that it’s planned to have Presearch as one if the 3 first partners to launch jukebox ! So it’s not really OK to disable.

**Comment by ashraffouda** [comment]
you can choose to deploy automatically or choose a farm
![Screenshot from 2021-09-07 10-31-14](https://user-images.githubusercontent.com/1689456/132312495-833fe5c6-e96c-4c9b-a08e-f06db93d96ce.png)
![Screenshot from 2021-09-07 10-32-17](https://user-images.githubusercontent.com/1689456/132312499-1309d310-9de2-453a-97ce-1e1318332b98.png)

**Comment by gmachtel** [comment]
True. I selected automatically but I don't know ofc whether the farm is already used for running a Presearch node or not. Can't we make sure that IPv4 address is not yet used for a Presearch node ?

**Comment by ashraffouda** [comment]
we can not check that, u need to check each presearch deployed node from its portal for that 
mybe the node @abdulgig tried is on the same farm

**Comment by gmachtel** [comment]
OK but a user that wants to set up a node has no view on which node has capacity reserved and which one hasn't. Can it be checked in the deployment process itself before selecting the node to deploy on ? In a VDC you're sure that it's available, there a user has the VDC for himself and can view all deployments that he made.

**Comment by ashraffouda** [comment]
away from the presearch issue, if you chose to deploy manually on a farm, the drop down that will appear contains only farms that have enough capacity

**Comment by gmachtel** [comment]
The issue here is not capacity as such, it’s only the question whether the same IPv4 address has already been used for other PRE deployments. Does it check on that as well ? 
we have to think from the user perspective: he will generally go for the easiest solution, which is automated farm selection. It should work without hassle in this case as well.

**Comment by ashraffouda** [comment]
I don't think it is a solution because using this approach we will be able to only deploy one presearch per node, I proposed solutions here to be discussed 
https://github.com/threefoldtech/tf_jukebox/issues/87

**Comment by ashraffouda** [comment]
actually if zos team can add ipv4 to grid2 containers this would be great, there was an issue for that but they closed it because in grid3 all deployments will go through vms but this is not ready now
